src/delfino/commands/typecheck.py

```diff
@@ -34,18 +23,66 @@ def typecheck(app_context: AppContext, summary_only: bool):
         "--warn-unused-config",
         "--warn-unused-ignores",
         "--color-output",
+        "--allow-untyped-decorators",
         "--follow-imports",
         "silent",
         "--junit-xml",
         reports_file,
-        delfino.sources_directory,
-        delfino.tests_directory,
     ]
 
-    if app_context.commands_directory.exists():
-        args.append(app_context.commands_directory)
+    if strict:
+        args.append("--strict")
+
+    args.extend(paths)
 
     if summary_only:
         args.extend(["|", "tail", "-n", "1"])
 
-    run(args, env_update_path={"MYPYPATH": delfino.sources_directory}, shell=summary_only, on_error=OnError.ABORT)
+    run(args, env_update_path={"MYPYPATH": mypypath}, on_error=OnError.ABORT)
+
+
+def is_path_relative_to_paths(path: Path, paths: List[Path]) -> bool:
+    for p in paths:
+        if path.is_relative_to(p):
+            return True
+    return False
+
+
+@click.command()
+@click.option("--summary-only", is_flag=True, help="Suppress error messages and show only summary error count.")
+@click.option(
+    "--path",
+    "-p",
+    multiple=True,
+    default=[],
+    help="Directory/File paths to type check. Overrides default path when passed.",
+)
```

Comment:
Is this options still necessary? What is the use case if so? For individual files/dirs, I would expect people to run tools directly rather than through delfino. So in this instance, `mypy <FILE>` rather than `delfino typecheck -p <FILE>`. If you think a path/file override would be useful, I would implement it in a more reusable fashion, which could be shared by multiple commands (for example linting, formatting, etc.). So that is is predictable and consistent.

Reply:
How about using [click.argument](https://click.palletsprojects.com/en/8.1.x/arguments/) instead of option?

The argument is more close to the original commands' (mypy/isort/black/pyilnt/pycodetyle) interface which takes an arbitrary number of paths.

With option(multiple=True);
```
delfino typecheck --path a.py --path b.py
```

With argument
```
delfino typecheck a.py b.py
```

The argument style is different from our existing internal invoke commands interface which supports `--path` option. But I think it won't be an issue as `--path` is for manual execution in our use case.

Reply:
I was thinking the same. It should still support additional options if needed. And I can't think of any other use case that we would like to use the argument for.